### ISSUE TITLE
Segmentation fault: 11 w/ --long-reads flag

### ISSUE THREAD

**Issue by phiweger** [bbuchfink/diamond#282]
I am trying to use `Diamond` to find proteins on some contigs (a `SPAdes` assembly), but it errs out w/ a segmentation fault:

```bash
diamond makedb --in proteins.faa --db proteins.dmnd
diamond blastx --threads 8 --db proteins.dmnd --query contigs.fasta --long-reads --outfmt 6 --out result.m8

diamond v0.9.24.125 | by Benjamin Buchfink <buchfink@gmail.com>
Licensed under the GNU GPL <https://www.gnu.org/licenses/gpl.txt>
Check http://github.com/bbuchfink/diamond for updates.

#CPU threads: 8
Scoring parameters: (Matrix=BLOSUM62 Lambda=0.267 K=0.041 Penalties=11/1)
Temporary directory: proteins
Opening the database...  [0.002166s]
Percentage range of top alignment score to report hits: 10
Opening the input file...  [8.1e-05s]
Opening the output file...  [2.5e-05s]
Loading query sequences...  [0.151964s]
Masking queries...  [0.859807s]
Building query seed set...  [0.042588s]
Algorithm: Double-indexed
Building query histograms...  [0.080687s]
Allocating buffers...  [2.6e-05s]
Loading reference sequences...  [0.056979s]
Building reference histograms...  [0.127855s]
Allocating buffers...  [2.2e-05s]
Initializing temporary storage...  [0.008961s]
Processing query chunk 0, reference chunk 0, shape 0, index chunk 0.
Building reference index...  [0.156116s]
Building query index...  [0.090257s]
Building seed filter...  [0.014748s]
Searching alignments...  [0.871728s]
Processing query chunk 0, reference chunk 0, shape 0, index chunk 1.
Building reference index...  [0.13826s]
Building query index...  [0.08805s]
Building seed filter...  [0.015442s]
Searching alignments...  [0.444482s]
Processing query chunk 0, reference chunk 0, shape 0, index chunk 2.
Building reference index...  [0.140574s]
Building query index...  [0.086939s]
Building seed filter...  [0.017964s]
Searching alignments...  [0.632058s]
Processing query chunk 0, reference chunk 0, shape 0, index chunk 3.
Building reference index...  [0.132979s]
Building query index...  [0.07899s]
Building seed filter...  [0.016538s]
Searching alignments...  [0.697344s]
Processing query chunk 0, reference chunk 0, shape 1, index chunk 0.
Building reference index...  [0.125917s]
Building query index...  [0.076574s]
Building seed filter...  [0.015275s]
Searching alignments...  [0.331923s]
Processing query chunk 0, reference chunk 0, shape 1, index chunk 1.
Building reference index...  [0.147705s]
Building query index...  [0.086453s]
Building seed filter...  [0.016585s]
Searching alignments...  [0.405867s]
Processing query chunk 0, reference chunk 0, shape 1, index chunk 2.
Building reference index...  [0.145957s]
Building query index...  [0.085412s]
Building seed filter...  [0.016245s]
Searching alignments...  [0.350217s]
Processing query chunk 0, reference chunk 0, shape 1, index chunk 3.
Building reference index...  [0.126605s]
Building query index...  [0.078394s]
Building seed filter...  [0.017907s]
Searching alignments...  [0.373614s]
Deallocating buffers...  [0.004859s]
Computing alignments... Segmentation fault: 11
```

Am I doing sth wrong or is this a bug?

Thanks!

**Comment by bbuchfink** [comment]
This is most likely a bug. Would it be possible to make your data available to me so I can try to reproduce it?

**Comment by phiweger** [comment]
I sent you the data in an email, thanks a lot.

**Comment by bbuchfink** [comment]
I reran the same command with your data but didn't get an error. Could you give me your system specs (CPU, memory, OS)? If you could try again on a different machine that would also help.

**Comment by phiweger** [comment]
Could this be a problem of not enough RAM?

**Comment by bbuchfink** [comment]
No, that should not produce that kind of error. If this is a MAC-specific problem it will be hard for me to help since I don't have access to a MAC.

**Comment by phiweger** [comment]
It seems Mac specific, bc/ I can successfully run the command using a Linux-based docker image. Well, I guess that'll work. Thanks anyway :)

**Comment by bbuchfink** [comment]
This is fixed in 9e8ec5a3d5eac17fbe17aee2117f3d9f595188f7.
